### ISSUE TITLE
htaccess Rewrite-Base brings up false positive security message

### ISSUE THREAD

**Issue by NitroxydeX** [nextcloud/server#2499]
### Steps to reproduce
1. enable 'htaccess.RewriteBase' in config
2. upgrade htaccess via command
3. get error message about accessible data directory

### Expected behaviour
No error message should appear

### Actual behaviour
Error message that data directory is accessible by direct access. False positive - requests will be redirected to login page. original data directory and original nextcloud htaccess in it.

### Server configuration
**Operating system**: Debian Jessie

**Web server:** Apache 2.4.10

**Database:** MYSQL 5.7.14

**PHP version:** 5.6.24-0+deb8u1

**Nextcloud version:** 10.0.1

**Updated from an older Nextcloud/ownCloud or fresh install:** Fresh Install

**Where did you install Nextcloud from:** Nextcloud Website archive

**Comment by Vlad-M-1** [comment]
I have the same problem with Nextcloud 10.0.1, Apache and Rewrite-Base.

The reason [is located there core/js/setupchecks.js](https://github.com/nextcloud/server/blob/v10.0.1/core/js/setupchecks.js#L201) :

```
		checkDataProtected: function() {
			var deferred = $.Deferred();
			if(oc_dataURL === false){
				return deferred.resolve([]);
			}
			var afterCall = function(xhr) {
				var messages = [];
				if (xhr.status !== 403 && xhr.status !== 307 && xhr.status !== 301 && xhr.responseText !== '') {
					messages.push({
						msg: t('core', 'Your data directory and your files are probably accessible from the Internet. The .htaccess file is not working. We strongly suggest that you configure your web server in a way that the data directory is no longer accessible or you move the data directory outside the web server document root.'),
						type: OC.SetupChecks.MESSAGE_TYPE_ERROR
					});
				}
				deferred.resolve(messages);
			};

			$.ajax({
				type: 'GET',
				url: OC.linkTo('', oc_dataURL+'/htaccesstest.txt?t=' + (new Date()).getTime()),
				complete: afterCall,
				allowAuthErrors: true
			});
			return deferred.promise();
},
```

As you can see, there's attempt to open test file. It's considered successfully open if response code is not 403, 307 and 301 and response body is not empty. The problem is Apache returns code 302 which is a redirect. This code is processed transparently, so we can't just check if `xhr.status` equals 302, as we only get 200 there.

I see 2 possible solutions there:
1. Change `xhr.responseText !== ''` to comparison with the actual text that should be in the test file `htaccesstest.txt`.
2. Find the RewriteRule that causes the redirect and add R=301 flag.

**Comment by MorrisJobke** [comment]
Maybe the same as https://github.com/nextcloud/server/issues/1051

@rullzer or @LukasReschke could one of you look into this false positive. I also have seen many reports for this in the forums.

**Comment by MorrisJobke** [comment]
I found the other ticket: #2499 

Closing as duplicate then.

**Comment by NitroxydeX** [comment]
@MorrisJobke Wrong ID? #2499 is this Ticket here.

**Comment by MorrisJobke** [comment]
> @MorrisJobke Wrong ID? #2499 is this Ticket here.

Yes - I meant #1364

And the fix is in #2513 ;)
